src/python_testing/TC_BOOLCFG_4_2.py

```diff
@@ -0,0 +1,176 @@
+from chip.interaction_model import InteractionModelError, Status
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+sensorTrigger = 0x0080_0000_0000_0000
+sensorUntrigger = 0x0080_0000_0000_0001
+
+
+class TC_BOOLCFG_4_2(MatterBaseTest):
+    async def read_boolcfg_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.BooleanStateConfiguration
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_BOOLCFG_4_2(self) -> str:
+        return "[TC-BOOLCFG-4.2] AlarmsActive attribute with DUT as Server"
+
+    def steps_TC_BOOLCFG_4_2(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep("2a", "Read FeatureMap attribute"),
+            TestStep("2b", "Read AttributeList attribute"),
+            TestStep("3a", "Create variable named enabledAlarms"),
+            TestStep("3b", "If VIS is supported, set bit 0 to 1"),
+            TestStep("3c", "If AUD is supported, set bit 1 to 1"),
+            TestStep("3d", "Set AlarmsEnabled attribute to value of enabledAlarms using EnableDisableAlarm command"),
+            TestStep(4, "Send TestEventTrigger with SensorTrigger event"),
+            TestStep(5, "Read AlarmsActive attribute"),
+            TestStep("6a", "Verify VIS alarm is active, if supported"),
+            TestStep("6b", "Verify VIS alarm is not active, if not supported"),
+            TestStep("7a", "Verify AUD alarm is active, if supported"),
+            TestStep("7b", "Verify AUD alarm is not active, if not supported"),
+            TestStep(8, "Send TestEventTrigger with SensorUntrigger event"),
+            TestStep(9, "Read AlarmsActive attribute"),
+        ]
+        return steps
+
+    def pics_TC_BOOLCFG_4_2(self) -> list[str]:
+        pics = [
+            "BOOLCFG.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_BOOLCFG_4_2(self):
+
+        asserts.assert_true('PIXIT.BOOLCFG.TEST_EVENT_TRIGGER_KEY' in self.matter_test_config.global_test_params,
+                            "PIXIT.BOOLCFG.TEST_EVENT_TRIGGER_KEY must be included on the command line in "
+                            "the --int-arg flag as PIXIT.BOOLCFG.TEST_EVENT_TRIGGER_KEY:<key>")
+
+        endpoint = self.user_params.get("endpoint", 1)
+        enableKey = self.matter_test_config.global_test_params['PIXIT.BOOLCFG.TEST_EVENT_TRIGGER_KEY'].to_bytes(16, byteorder='big')
+
+        self.step(1)
+        attributes = Clusters.BooleanStateConfiguration.Attributes
+
+        self.step("2a")
+        feature_map = await self.read_boolcfg_attribute_expect_success(endpoint=endpoint, attribute=attributes.FeatureMap)
+
+        is_vis_feature_supported = feature_map & Clusters.BooleanStateConfiguration.Bitmaps.Feature.kVisual
```

Comment:
```suggestion
        is_VIS_feature_supported = feature_map & Clusters.BooleanStateConfiguration.Bitmaps.Feature.kVisual
```

Reply:
I get this, but I would prefer not to mix case if that is ok?